ir/src/symbol_table.rs

```diff
@@ -69,6 +69,11 @@ impl SymbolTable {
         &mut self,
         columns: &[Identifier],
     ) -> Result<(), SemanticError> {
+        if columns.is_empty() {
+            return Err(SemanticError::MissingMainTraceColumns(
+                "Main Trace Columns not defined".to_string(),
+            ));
+        }
```

Comment:
I think this check should be handled in the parser instead. It should be easy to add it there, since there's already an error raised when both trace section definitions are missing

Reply:
Fixed, thanks. There is one other issue. The parser expects the main trace columns to be defined before aux trace columns. Should we allow different ordering? If not, maybe the error should be more descriptive. I've opened #55 for this.

Reply:
@tohrnii thanks for noticing and addressing this. Note for the future - the resolution is for this ordering to be required and to make an adjustment to the docs, which is in #36